src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java

```diff
@@ -551,6 +551,29 @@ public void given_an_event_when_directionDueDate_is_not_set_then_task_should_sta
         taskToTearDown = taskId;
     }
 
+    @Test
+    public void given_initiation_task_with_followup_overdue_ctg_when_cancelled_with_noc_event_then_cancel_the_task() {
+        String caseId1 = UUID.randomUUID().toString();
+        String taskIdDmnColumn = "followUpOverdueCaseBuilding";
+
+        // caseId1 with category Followup overdue
+        // task1
+        final String caseId1Task1Id = initiateTaskForGivenId(
+            caseId1,
+            "requestCaseBuilding",
+            "", "caseBuilding", false,
+            taskIdDmnColumn
+        );
+
+        // Then cancel all tasks on both caseIDs
+        sendMessage(caseId1, "applyNocDecision", "", "", false);
+        waitSeconds(5);
```

Comment:
don't think we need the wait if we are using the pulling mechanism

Reply:
There wasn't any . But added one.

Reply:
Nice one Vam 🚀